src/ui/textfield.rs

```diff
@@ -1,15 +1,63 @@
 //! A fairly simplistic text field implementation.
 
+use std::{borrow::BorrowMut, ops::Range};
+
+use copypasta::{ClipboardContext, ClipboardProvider};
 use skulpin::skia_safe::*;
 
 use crate::ui::*;
 
+/// Text field selection struct
+struct Selection {
+    pub first: usize,
+    pub second: usize,
```

Comment:
I think the names of these fields could better summarize what they're for; maybe `first` → `cursor`, and `second` → `anchor`? Then they could also use some doc comments, like:
```rs
/// The actual text cursor as moved by the user, shown as a black (or white) beam in the text.
cursor: usize,
/// The "selection anchor". This is where the selection is anchored at.
/// This anchor normally moves together with the cursor, but becomes detached once the user presses Shift.
anchor: usize,
```
Also, the fields don't need to be public as `struct Selection` is private to the textfield module.